src/lib/containers/JiraIssueCollector.tsx

```diff
@@ -0,0 +1,164 @@
+}
+
+declare global {
+  interface Window {
+    jQuery?: JQueryStatic
+    ATL_JQ_PAGE_PROPS: any
+  }
+}
+
+type JiraIssueCollectorConfig = {
+  issueCollector: IssueCollector
+  issueSummary: string
+  issueDescription: string
+  issuePriority: '1' | '2' | '3' | '4' | '5'
+  accessRequirementId?: string
+  synapseDataObjectId?: string
+  componentId?: string
+  principalId: string
+  userDisplayName?: string
+  userEmailAddress?: string
+}
+
+/**
+ * Shows the Jira issue collector when invoked.
+ * @param config
+ */
+export function showJiraIssueCollector(config: JiraIssueCollectorConfig): void {
+  const {
+    issueCollector,
+    issueSummary,
+    issueDescription,
+    principalId,
+    userDisplayName,
+    userEmailAddress,
+    synapseDataObjectId,
+    componentId,
+    accessRequirementId,
+    issuePriority,
+  } = config
+  try {
+    if (window.jQuery == null) {
+      throw new Error(
+        'jQuery is not present. Jira Issue Collector will not be shown.',
+      )
+    }
+    // Requires jQuery!
+    window.jQuery.ajax({
+      url: ISSUE_COLLECTOR_MAP[issueCollector],
+      type: 'get',
+      cache: true,
+      dataType: 'script',
+    })
+
+    window.ATL_JQ_PAGE_PROPS = {
+      triggerFunction: function (showCollectorDialog: () => void) {
+        showCollectorDialog()
+      },
+
+      fieldValues: {
+        summary: issueSummary,
+        description: issueDescription,
+        priority: issuePriority,
+        customfield_10840: userEmailAddress,
+        email: userEmailAddress,
+        customfield_10841: accessRequirementId,
+        customfield_10740: principalId,
+        customfield_10741: userDisplayName,
+        customfield_10742: synapseDataObjectId,
+        fullname: userDisplayName,
+        components: componentId,
+        // Component ID of the component added to the Jira Governance Project
+      },
+    }
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+type UseJiraIssueCollectorConfig = Omit<
+  JiraIssueCollectorConfig,
+  'principalId' | 'userDisplayName' | 'userEmailAddress'
+> & {
+  show: boolean
+}
+
+/**
+ * Hook used to show the Jira issue collector within Synapse. The collector will only be shown if the user is logged in.
+ * @requires SynapseContext to be an ancestor in the component tree
+ */
+export function useJiraIssueCollector(config: UseJiraIssueCollectorConfig) {
+  const {
+    show,
+    issueCollector,
+    issueSummary,
+    issueDescription,
+    issuePriority,
+    accessRequirementId,
+    synapseDataObjectId,
+    componentId,
+  } = config
+
+  const { accessToken } = useSynapseContext()
+
+  const { data: currentUser, isLoading: profileIsLoading } =
+    useGetCurrentUserProfile({
+      enabled: !!accessToken,
+    })
+  const { data: notificationEmail, isLoading: emailIsLoading } =
+    useGetNotificationEmail({
+      enabled: !!accessToken,
+    })
+
+  useEffect(() => {
+    if (show && accessToken) {
```

Comment:
Should we show the issue collector for anonymous users?

Reply:
Well, I'd prefer to hear from Synapse users so we can follow up for more information (if necessary).  On the other hand, we'd probably rather get too many reports than too few.